src/modules/flake-parts/implementation.nix

```diff
@@ -0,0 +1,49 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  l = lib // builtins;
+  d2n = config.dream2nix;
+  makeArgs = p:
+    {
+      inherit (config) systems;
+      inherit (d2n) config;
+    }
+    // p;
+  outputs = d2n.lib.dlib.mergeFlakes (
+    l.map
+    (p: d2n.lib.makeFlakeOutputs (makeArgs p))
+    (l.attrValues d2n.inputs)
+  );
+in {
+  config = {
+    flake =
+      # make attrs default, so that users can override them without
+      # needing to use lib.mkOverride (usually, lib.mkForce)
+      l.mapAttrsRecursiveCond
+      d2n.lib.dlib.isNotDrvAttrs
+      (_: l.mkDefault)
+      outputs;
+    dream2nix.outputs = outputs;
+    perSystem = {
+      config,
+      system,
+      ...
+    }: let
+      # get output attrs that have systems
+      systemizedOutputs =
+        l.mapAttrs
+        (_: attrs: attrs.${system})
+        (
+          l.filterAttrs
+          (_: attrs: l.isAttrs attrs && l.hasAttr system attrs)
+          outputs
+        );
+    in {
+      config = {
+        dream2nix.outputs = systemizedOutputs;
```

Comment:
Are these flake-style outputs? You could assign them to the options so that they're automatically exported.

Reply:
Those are outputs that don't have a `system` yes. I didn't assign them here since we already export everything that was generated before in `config.flake` option. But kept this here in case people want to directly use any of the outputs that was generated by dream2nix.

Reply:
Ah, that's true, I forgot about that, I thought `flake-parts` would pass the stuff set in `config.flake` to the `perSystem` options when possible. I will push a fix for that :+1: